exporter/collector/metricsexporter.go

```diff
@@ -215,14 +286,21 @@ func (me *MetricsExporter) projectName() string {
 }
 
 // Helper method to send metric descriptors to GCM.
-func (me *MetricsExporter) exportMetricDescriptor(ctx context.Context, md *metricpb.MetricDescriptor) error {
-	// export
+func (me *MetricsExporter) exportMetricDescriptor(md *metricpb.MetricDescriptor) {
+	if _, exists := me.mdCache[md.Type]; exists {
+		return
+	}
+	me.mdCache[md.Type] = md
```

Comment:
should we only set this if we were successful in creating the metric descriptor?

Reply:
Hmm. I guess that depends on whether we believe the error to be ephemeral or not.